Source/Core/Core/PowerPC/MMU.h

```diff
@@ -15,30 +15,115 @@ namespace PowerPC
 {
 // Routines for debugger UI, cheats, etc. to access emulated memory from the
 // perspective of the CPU.  Not for use by core emulation routines.
-// Use "Host_" prefix.
+// Use "Host" prefix.
+
+enum class RequestedAddressSpace
+{
+  Effective,  // whatever the current MMU state is
+  Physical,   // as if the MMU was turned off
+  Virtual,    // specifically want MMU turned on, fails if off
+};
+
+// Reads a value from emulated memory using the currently active MMU settings.
+// If the read fails (eg. address does not correspond to a mapped address in the current address
+// space), a PanicAlert will be shown to the user and zero (or an empty string for the string case)
+// will be returned.
 u8 HostRead_U8(u32 address);
 u16 HostRead_U16(u32 address);
 u32 HostRead_U32(u32 address);
 u64 HostRead_U64(u32 address);
 float HostRead_F32(u32 address);
 double HostRead_F64(u32 address);
 u32 HostRead_Instruction(u32 address);
+std::string HostGetString(u32 address, size_t size = 0);
+
+template <typename T>
+struct TryReadResult
+{
+  bool success;     // whether the read succeeded; if false, the other fields should not be touched
```

Comment:
Could you put this comment above `success` (and same for the other members)? Otherwise, clang-format will try to align all these comments which means that any change that modifies the column width of the members will result in a very noisy diff.

Reply:
Done.